src/Microsoft.AspNet.Cors.Core/CorsService.cs

```diff
@@ -137,7 +137,9 @@ public virtual void ApplyResult(CorsResult result, HttpResponse response)
 
                 if (nonSimpleAllowMethods.Length > 0)
                 {
-                    headers.Add(CorsConstants.AccessControlAllowMethods, nonSimpleAllowMethods);
+                    headers.SetCommaSeparatedValues(
```

Comment:
If you want them to end up as separate headers you need to call SetValues:
https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http.Abstractions/IHeaderDictionary.cs#L75